src/components/Card.js

```diff
@@ -1,9 +1,21 @@
 export default class Card {
-  constructor(data, cardSelector, handleCardClick) {
+  constructor({
+    data,
+    cardSelector,
+    handleCardClick,
+    handleDeleteIcon,
+    confirmPopup,
+    api,
+  }) {
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteIcon = handleDeleteIcon;
+    this._confirmPopup = confirmPopup;
+    this._api = api;
```

Comment:
Needs correction: the Card class should not know anything about the api. You should make the requests in index.js